libraries/ui/src/CourseCard.test.tsx

```diff
@@ -0,0 +1,37 @@
+import { describe, expect, test } from 'vitest';
+import { render } from '@testing-library/react';
+import { CourseCard } from './CourseCard';
+
+describe('CourseCard', () => {
+  const baseProps = {
+    title: 'Introduction to Testing',
+    description: 'Learn proper testing techniques',
+    courseType: 'Crash course',
+    image: '/test-image.jpg'
+  };
+
+  test('renders default as expected', () => {
+    const { container } = render(<CourseCard {...baseProps} />);
+    expect(container).toMatchSnapshot();
+  });
+
+  test('renders with optional args', () => {
+    const { container } = render(
+      <CourseCard
+        {...baseProps}
+        className="custom-class"
+        cardType="Featured"
+      />
+    );
+    expect(container).toMatchSnapshot();
+  });
+
+  test('renders with optional yield', () => {
+    const { container } = render(
+      <CourseCard {...baseProps}>
+        <button>Enroll Now</button>
+      </CourseCard>
+    );
+    expect(container).toMatchSnapshot();
```

Comment:
this definitely works!, but instead of outputting the entire snapshot for this specific test, id recommend we just do an existence check for the yielded content, as that's all we really need to confirm here:
```suggestion
    const yieldedContent = screen.getByTestId('test-element');
    expect(yieldedContent).toBeTruthy();
```
you can reference [libraries/ui/src/CTALinkOrButton.test.tsx](https://github.com/bluedotimpact/bluedot/blob/baa3d4bdd7e0ea216f8156cae0378d92112b423e/libraries/ui/src/CTALinkOrButton.test.tsx#L4) as a similar example